Paws4Me/Paws4Me/Data Classes/ValidationManager.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  ValidationManager.swift
+//  Paws4Me
+//
+//  Created by Kirsty-Lee Walker on 2022/05/16.
+//
+
+import Foundation
+import UIKit
+
+class ValidationManager {
+
+    func isValidName(name: String) -> Bool {
+        var isNameValid = true
+        let nameRegex =  "[a-zA-Z]{2,}"
+          do {
+              let regex = try NSRegularExpression(pattern: nameRegex)
+              let nsString = name as NSString
+              let results = regex.matches(in: name, range: NSRange(location: 0, length: nsString.length))
+
+              if results.count == 0 {
+                  isNameValid = false
+              }
+          } catch {
+              isNameValid = false
+          }
+          return  isNameValid
+      }
+
+    func isValidSurname(surname: String) -> Bool {
+        var isSurnameValid = true
+        let surnameRegex =  "[a-zA-Z]{2,}"
+          do {
+              let regex = try NSRegularExpression(pattern: surnameRegex)
+              let nsString = surname as NSString
+              let results = regex.matches(in: surname, range: NSRange(location: 0, length: nsString.length))
+
+              if results.count == 0 {
+                  isSurnameValid = false
+              }
+          } catch {
+              isSurnameValid = false
+          }
+          return  isSurnameValid
+      }
+
+    func isValidEmailAddress(emailAddress: String) -> Bool {
```

Comment:
This code can be simplified quite heavily, please revise based on the bottom 😄 also check out the naming convention 👍 

```
func isValidEmail(_ email: String) -> Bool {        
    let emailRegEx = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"

    let emailPred = NSPredicate(format:"SELF MATCHES %@", emailRegEx)
    return emailPred.evaluate(with: email)
}
```

Reply:
I did see this solution on GitHub, thank you Brandon. It is much simpler to use NSPredicate 👍🏻